File: 5. LinkedList/deleteFirstNode.java
package Linked_revision;

public class deleteFirstNode {
	public static Node<Integer> createLinkedList()
	{
		Node<Integer> n1 = new Node<>(10);
		Node<Integer> n2 = new Node<>(20);
		Node<Integer> n3 = new Node<>(30);
		Node<Integer> n4 = new Node<>(40);
		n1.next = n2;
		n2.next = n3;
		n3.next = n4;
		return n1;
	}
	public static void printLinkedList(Node<Integer> head)
	{
		while(head != null)
		{
			System.out.println(head.data);
			head = head.next;
		}
	}
	
	public static Node<Integer> deleteAtFirst(Node<Integer> head)
	{
		Node<Integer> newhead = head.next;
		return newhead;
	}
	
	public static void main(String[] args)
	{
		Node<Integer> head = createLinkedList();
		System.out.println("Before Deletion : ");
		printLinkedList(head);
		System.out.println("After Deletion : ");
		head = deleteAtFirst(head);
		printLinkedList(head);
	}
}

File: Yash/C++/DSA/3. LinkedList/deleteFirstNode.cpp
#include <iostream>
using namespace std;

class Node{
    public:
    int data;
    Node *next;
    Node(int data)
    {
        this->data = data;
        next = nullptr;
    }
};

Node *createLinkedList()
{
    Node *a1 = new Node(10);
    Node *a2 = new Node(20);
    Node *a3 = new Node(30);
    Node *a4 = new Node(40);
    Node *a5 = new Node(50);
    a1->next = a2;
    a2->next = a3;
    a3->next = a4;
    a4->next = a5;

    return a1;    
}

void printLinkedList(Node *head)
{
    Node *temp = head;
    while(temp!= nullptr)
    {
        cout<<temp->data<<endl;
        temp = temp->next;
    }
}

Node *deleteFirstNode(Node *head)
{
    head = head->next;
    return head;
}


int main()
{
    Node *head = createLinkedList();
    cout<<"Before deleting "<<endl;
    printLinkedList(head);
    cout<<"After deleting "<<endl;
    Node *new_head = deleteFirstNode(head);
    printLinkedList(new_head);
    return 0;
}